### PR TITLE
[MTDSA-8548] NRS Connector & Service for NRS in FP BSAS

### PR DESCRIPTION
**Pull request by Mbrookesy**
Includes a few renaming of the SE NRS connector and service

### CODE REVIEW

File: test/v2/services/SubmitForeignPropertyBsasNrsProxyServiceSpec.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+
+  "NrsProxyService" should {
+    "call the Nrs Proxy connector" when {
+      "the connector is valid" in new Test {
+
+        MockNrsProxyConnector.submit(nino.toString())
+          .returns(Future.successful((): Unit))
+
+        service.submit(nino.toString(), SubmitForeignPropertyBsasRequestBody(None, None))
```
Comment (james-work-account) [inline test/v2/services/SubmitForeignPropertyBsasNrsProxyServiceSpec.scala:41]: This isn't a test - you're just running the service function and not comparing it to anything.

Comment (james-work-account) [reply test/v2/services/SubmitForeignPropertyBsasNrsProxyServiceSpec.scala:41]: If the other service/s is/are doing the same, it/they needs/need changing too.

File: app/v2/services/SubmitForeignPropertyBsasNrsProxyService.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import javax.inject.Inject
+import uk.gov.hmrc.http.HeaderCarrier
+import v2.connectors.SubmitForeignPropertyBsasNrsProxyConnector
+import v2.models.request.submitBsas.foreignProperty.SubmitForeignPropertyBsasRequestBody
+
+import scala.concurrent.ExecutionContext
+
+class SubmitForeignPropertyBsasNrsProxyService @Inject()(val connector: SubmitForeignPropertyBsasNrsProxyConnector) {
+
+  def submit(nino: String, body: SubmitForeignPropertyBsasRequestBody)(implicit hc: HeaderCarrier, ec: ExecutionContext): Unit = {
```
Comment (james-work-account) [inline app/v2/services/SubmitForeignPropertyBsasNrsProxyService.scala:28]: Change return type to `Future[Unit]`

**Review by james-work-account — CHANGES_REQUESTED** [review]

File: app/v2/services/SubmitForeignPropertyBsasNrsProxyService.scala (hunk 2 above)
Comment (james-work-account) [reply app/v2/services/SubmitForeignPropertyBsasNrsProxyService.scala:28]: Same for any other services

**Review by Nboaram — APPROVED** [review]

**Pull request merged**